### PR TITLE
Add support for 'Add' in AskNonPositiveHandler and AskNonNegativeHandler

### PR DESCRIPTION
**Pull request by debugger22**
Although it's a positive step, but it broke the following

```
    assert ask(Q.positive(asin(x)), Q.positive(x) & Q.nonpositive(x - 1)) is True
    assert ask(Q.positive(asin(x)), Q.negative(x) & Q.nonnegative(x + 1)) is False
```

@asmeurer I can't figure out what's wrong here.

### CODE REVIEW

**Comment by asmeurer** [comment]
This code style is a little hard to read.

**Comment by asmeurer** [comment]
This is the problem 

```
In [3]: ask(Q.nonpositive(x - 1), Q.positive(x))
Out[3]: False
```

The problem is that the outermost `fuzzy_and` is wrong. If all are True, you can make the deduction, but if one is False, it should give None, not False. I think there is a closed_group function that can help here.

File: sympy/assumptions/handlers/order.py [hunk 1]
```diff
@@ -134,6 +134,21 @@ def Basic(expr, assumptions):
                 return ask(Q.real(expr), assumptions)
             else:
                 return notnegative
+        if expr.is_Add:
```
Comment (asmeurer) [inline sympy/assumptions/handlers/order.py]: Don't do a type check here. Add a staticmethod for `Add`.

**Comment by asmeurer** [comment]
Or rather, this should mimic (and probably reuse code from) the negative and positive Add handlers.

**Comment by asmeurer** [comment]
You could also just do `nonpositive = real & not positive`. This will limit some deductions you can get vs. the explicit way, but the only way to get every deduction is to implement it in satask.

**Comment by czgdp1807** [comment]
The following fails,
```python
assert ask(Q.positive(acos(x)), Q.zero(x)) is True
```

**Comment by czgdp1807** [comment]
Closing for now. Thanks for your contributions.

**Pull request closed without merging**